fastify.js

```diff
@@ -417,15 +417,19 @@ function build (options) {
   }
 
   function _route (_fastify, method, url, options, handler) {
-    if (!handler && typeof options === 'function') {
-      handler = options
-      options = {}
+    if (!handler) {
+      if (typeof options === 'function') {
+        handler = options
+        options = {}
+      } else if (Object.prototype.toString.call(options) !== '[object Object]') {
+        throw new TypeError(`options for route ${method}:${url} must be an object`)
+      }
```

Comment:
The point of the special case for `handler` below was to remove the need for this extra code, so this part can be reverted.

Reply:
Aye, there is no need for this part part.